pact/Pact/Core/IR/Eval/CEK.hs

```diff
@@ -404,9 +405,7 @@ applyPact i pc ps cont handler cenv nested = useEvalState esDefPactExec >>= \cas
       -- `initPact` ensures that the step is 0,
       -- and there are guaranteed more than 0 steps due to how the parser is written.
       -- `resumePact` does a similar check before calling this function.
-      when (ps ^. psStep >= nSteps) $ failInvariant i "Step not found"
```

Comment:
Why this change?

Reply:
Great question. It's because If you notice below, we actually access the steps partially via a lens and throw an error using `maybe`. It's sort of redundant to throw this invariant error twice.